PhysicsTools/PatAlgos/plugins/GenMETExtractor.h

```diff
@@ -0,0 +1,45 @@
+
+#ifndef PhysicsTools_PatAlgos_GenMETExtractor_h
+#define PhysicsTools_PatAlgos_GenMETExtractor_h
+
+/**
+  \class    pat::GenMETExtractor GenMETExtractor.h "PhysicsTools/PatAlgos/interface/GenMETExtractor.h"
+  \brief    Retrieves the genMET from a pat::MET
+
+   The GenMETExtractor produces the analysis-level pat::MET starting from
+   a collection of objects of METType.
+
+  \author   Matthieu Marionneau
+  \version  $Id: GenMETExtractor.h,v 1.0 2015/07/22 mmarionn Exp $
+*/
+
+
+#include "FWCore/Framework/interface/EDProducer.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/ConsumesCollector.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Utilities/interface/InputTag.h"
+
+#include "DataFormats/PatCandidates/interface/MET.h"
+#include "DataFormats/METReco/interface/GenMET.h"
+
+namespace pat {
+
+  class GenMETExtractor : public edm::EDProducer {
```

Comment:
Given that no member data are modified during the call to ```produce```, this should be changed to an ```edm::global::EDProducer<>```. However, that means the signature of produce has to change to 
```void produce(edm::StreamID, edm::Event&, edm::EventSetup const&) const```

See https://twiki.cern.ch/twiki/bin/view/CMSPublic/FWMultithreadedFrameworkGlobalModuleInterface#Base_Class for details